python-lib/gluonts_forecasts/model_handler.py

```diff
@@ -0,0 +1,193 @@
+    pass
+
+
+class ModelHandler:
+    """
+    Class to retrieve the estimator, trainer or descriptor of a GluonTS model
+
+    Attributes:
+        model_name (str)
+    """
+
+    def __init__(self, model_name):
+        self.model_name = model_name
+        self.model_descriptor = self._get_model_descriptor()
+
+    def _get_model_descriptor(self):
+        model_descriptor = MODEL_DESCRIPTORS.get(self.model_name)
+        if model_descriptor is None:
+            return MODEL_DESCRIPTORS.get("naive")
+        else:
+            return model_descriptor
+
+    def estimator(self, model_parameters, **kwargs):
+        kwargs.update(model_parameters.get("kwargs", {}))
+        estimator = self.model_descriptor.get(ESTIMATOR)
+        try:
+            ret = None if estimator is None else estimator(**kwargs)
+        except Exception as err:
+            raise ModelParameterError("Issue with the parameters ({}) for model '{}' estimator: {}".format(kwargs, self.model_name, err))
+        return ret
+
+    def trainer(self, **kwargs):
+        trainer = self.model_descriptor.get(TRAINER)
+        try:
+            ret = None if trainer is None else trainer(**kwargs)
+        except Exception as err:
+            raise ModelParameterError("Issue with the parameters ({}) for model '{}' trainer: {}".format(kwargs, self.model_name, err))
+        return ret
+
+    def predictor(self, **kwargs):
+        predictor = self.model_descriptor.get(PREDICTOR)
+        try:
+            ret = None if predictor is None else predictor(**kwargs)
+        except Exception as err:
+            raise ModelParameterError("Issue with the parameters ({}) for model '{}' predictor: {}".format(kwargs, self.model_name, err))
+        return ret
+
+    def can_use_external_feature(self):
+        return self.model_descriptor.get(CAN_USE_EXTERNAL_FEATURES, False)
+
+    def can_use_context_length(self):
+        return self.model_descriptor.get(CAN_USE_CONTEXT_LENGTH, True)
+
+    def needs_num_samples(self):
+        return self.model_descriptor.get(NEEDS_NUM_SAMPLES, False)
+
+    def get_label(self):
+        return self.model_descriptor.get(LABEL, "")
+
+
+def list_available_models():
+    """List available models names found in the recipe.json (keys of MODEL_DESCRIPTORS).
+
+    Returns:
+        dict_keys of model names.
+    """
+    available_models = MODEL_DESCRIPTORS.copy()
+    return available_models.keys()
+
+
+def list_available_models_labels():
+    """List available models labels found in the UI.
+
+    Returns:
+        List of model names.
+    """
+    available_models = MODEL_DESCRIPTORS.copy()
+    available_models_labels = []
+    for model in available_models:
+        label = available_models[model].get(LABEL)
+        if label is not None:
+            available_models_labels.append(label)
+    return available_models_labels
+
+
+def list_naive_models_labels():
```

Comment:
better to have a get_naive_models_descriptors (implemented with filter + lambda) and a get_naive_model_label which call this method and do a map on it.